File: app/src/main/java/com/example/kotlin_demo/MainActivity.java
package com.example.kotlin_demo;

import android.app.Activity;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main); // Ensure you have activity_main.xml in res/layout
    }
}

File: app/src/main/java/com/example/kotlin_demo/MainActivity.kt
package com.example.kotlin_demo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.LifecycleObserver

class MainActivity : AppCompatActivity() , LifecycleObserver {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
    }
}